qutip/core/qobjevo.py

```diff
@@ -586,32 +632,45 @@ def __neg__(self):
         res.cte = -res.cte
         for op in res.ops:
             op.qobj = -op.qobj
-        res.compile()
+        res._compile()
         return res
 
+    def __and__(self, other):
+        """
+        Syntax shortcut for tensor:
+        A & B ==> tensor(A, B)
+        """
+        return qutip.tensor(self, other)
```

Comment:
if you import `tensor` as a trailing import at the end, you don't need to prefix it with `qutip` and that helps the core package keep its separability from the rest of QuTiP (though I'd still like to break more of the circular imports) - like is done in `qobj.py`

I'm sure there's a nicer way to handle these circular imports anyway, I just haven't thought it through properly yet.  I _used_ to have to be much stricter about it back when I wrote F#...

Reply:
There are a few circular import created by this `&` and `Coefficient * Qobj` product... 
The only way I see to remove all these would be to monkey patch this `__and__` method in tensor.py.

Reply:
Yeah, I suppose it's not an easy problem to solve right now.  I have a (slightly irrational) dislike of circular imports and of subpackages not having a strict order (e.g. `qutip.core` should ideally be fully initialised before any other part of `qutip` is accessed), but it's not a big deal right now.